src/Arcus.WebApi.Security.Authorization/AzureManagedIdentityAuthorizationOptions.cs

```diff
@@ -0,0 +1,20 @@
+﻿using GuardNet;
+
+namespace Arcus.WebApi.Security.Authorization
+{
+    public class AzureManagedIdentityAuthorizationOptions
+    {
+        private const string DefaultHeaderName = "x-managed-identity-token";
+
+        public string HeaderName { get; set; }
```

Comment:
```suggestion
        public string HeaderName { get; }
```

Reply:
If you do that, people won't be able to change the default value to whatevery they want to use

Reply:
Hmm, weird.  I can commit my own suggestions :)

Reply:
I'd say you'd do that (chaning the headername) by using the constructor ?

Reply:
Ah, ok now I see.
You want it to behave like this:
```csharp
.AddAuthorizationFilter( o => o.HeaderName = "foo");
```
Ok then...

Reply:
So you'd rather want to create the object yourself rather than a fluent model?

See https://github.com/arcus-azure/arcus.webapi/pull/147/files#diff-3399f8d754d976ba2fdf249b575dff02R91 for example

Reply:
Yes, I think I've the same idea as @fgheysels to use the object here since it's more clear what filter you gonna add + it's the current appraoch on all filters in the project.

Reply:
> Yes, I think I've the same idea as @fgheysels to use the object here since it's more clear what filter you gonna add + it's the current appraoch on all filters in the project.

Can you elaborate on that? Not sure if I'm following.

You're saying we should always go over ctor? If so, what do you propose for a consumer friendly appraoch of configuring it?

Reply:
Bc we're currently using that type-approach, I'd say yes, favoring the ctor, but that can be injected. We can pass along the time and call the `.Configure` on the service collection later. (Like the .NET startup is mostly set up I guess).

Reply:
But we can of course provide such an extension, it just seems more in line with the other built-in components that you 'register' the type somewhere and 'configure' it somewhere else.

Reply:
While that's true, I'm wondering if that is the approach we want to use.

I'd rather have our users do: `filter.AddXXX` instead of having to do two calls, which they might miss in the docs and have to provide support on it.

But it might be just me.

Reply:
Yes, I know. It's a bit discovery against user-friendly. Too much extensions make it not discoverable. While two or sometimes even more is not very user-friendly.
Can we provide both? Make the options and filter public enough to be used in both contexts and provide and extension that do both things?

Reply:
Although I very much like immutable types, I think we should go with the way that is 'generally accepted' in asp.net, and that is the initial proposal of Tom:

Something like this:

```csharp
.AddAuthorizationFilter(options => options.HeaderName = "myFooBarHeader");
```